pkg/api/v1alpha1/vspheredatacenterconfig_webhook.go

```diff
@@ -74,6 +81,16 @@ func (r *VSphereDatacenterConfig) ValidateUpdate(old runtime.Object) error {
 	if !ok {
 		return apierrors.NewBadRequest(fmt.Sprintf("expected a VSphereDataCenterConfig but got a %T", old))
 	}
+	var allErrs field.ErrorList
+
+	if err := r.Validate(); err != nil {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec"), r.Spec, err.Error()))
+		return apierrors.NewInvalid(
+			GroupVersion.WithKind(VSphereDatacenterKind).GroupKind(),
+			r.Name,
+			allErrs,
+		)
+	}
```

Comment:
```suggestion
	if err := r.Validate(); err != nil {
		return apierrors.NewInvalid(
		           GroupVersion.WithKind(VSphereDatacenterKind).GroupKind(),
		           r.Name,
			       field.ErrorList{
			           field.Invalid(field.NewPath("spec"), r.Spec, err.Error())
			       },
			   )
	}
```

Reply:
we append to the error list on line 91 below so didnt end up removing the error list declaration.